libalpm/scripts/mkinitcpio-remove

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -e
+
+package=0
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
+        # triggers when it's a change to usr/lib/initcpio/*
+        package=1
+        continue
+    fi
+
+    if ! read -r pkgbase > /dev/null 2>&1 < "${line%/vmlinuz}/pkgbase"; then
+        # if the kernel has no pkgbase, we skip it
+        continue
+    fi
+
+    # remove the actual kernel and images for the package being removed
+    kernel="/boot/vmlinuz-${pkgbase}"
+    preset="/etc/mkinitcpio.d/${pkgbase}.preset"
+    initramfs="/boot/initramfs-${pkgbase}.img"
+    fallback_initramfs="/boot/initramfs-${pkgbase}-fallback.img"
+    if [[ -e $kernel ]]; then
+        # remove the installed kernel
+        rm $kernel
+    fi
+    if [[ -e $preset ]]; then
+        # remove the preset
+        rm $preset
```

Comment:
So we're back to deleting userdata? When this file was a pacman-managed configuration file, uninstalling a kernel would result in this becoming a pacsave, so you could reinstall the kernel and restore your custom configuration. Now deleting a kernel causes your modifications in /etc to be magically nuked.

Reply:
Yes, for now. I'm going to rework this script once I add the option for mkinitcpio to remove the images. I'm going to add a check if this file was changed, by comparing it with one generated from the preset, and if it's different, I'll move it to .pacsave.

Reply:
I still think just leaving the file in place is much easier than trying to duplicate the backup algorithm,

Reply:
@heftig Trying to take over pacman's job means you need to take over pacman's job all the way. Leaving the hook behind means the preset is broken, and thus, `mkinitcpio --allpresets` is broken

Reply:
@grazzolini Another solution is kernels can just... install them themselves?

It's an inert file, it does no harm if mkinitcpio is not installed... I don't understand why core/linux is trying so hard to strip itself of usefulness and make literally everything someone else's responsibility right down to actually installing the vmlinuz file somewhere usable.

Reply:
@eli-schwartz You're right; I forgot about `mkinitcpio -P`.

Reply:
Honestly if the worry is about duplicating the preset template file we could still install it with mkinitcpio and have the kernel makedepend on mkinitcpio to use that.

Reply:
> @grazzolini Another solution is kernels can just... install them themselves?
> 
> It's an inert _configuration_ file, it does no harm if mkinitcpio is not installed since nothing reads it... I don't understand why core/linux is trying so hard to strip itself of usefulness and make literally everything someone else's responsibility right down to actually installing the vmlinuz file somewhere usable.

So, will the kernels also install a dracut configuration for each of them? I agree with the decision of removing anything mkinitcpio specific from the kernels. It was the right move.

Reply:
I lack knowledge of how dracut operates, so I don't know if there's a comparable option. But sure, why not? There are absolutely tons of packages which install optional components that are only used by another package that may not be installed.

For example:

Lots of packages install /usr/lib/initcpio/ files, whether or not mkinitcpio is installed. These files do nothing, except advertise functionality if mkinitcpio is installed or is later installed.


Some packages do the same for dracut:

```
$ pkgfile -vr /usr/lib/dracut| grep -v ^extra/dracut
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/02zfsexpandknowledge/module-setup.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/export-zfs.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/module-setup.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/mount-zfs.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/parse-zfs.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/zfs-env-bootfs.service
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/zfs-generator.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/zfs-lib.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/zfs-load-key.sh
eschwartz/zfs-utils 0.8.2-2	/usr/lib/dracut/modules.d/90zfs/zfs-needshutdown.sh
community/clevis 11-6      	/usr/lib/dracut/modules.d/60clevis/clevis-hook.sh
community/clevis 11-6      	/usr/lib/dracut/modules.d/60clevis/module-setup.sh
community/nvme-cli 1.9-1   	/usr/lib/dracut/dracut.conf.d/70-nvmf-autoconnect.conf
community/tpm2-totp 0.2.0-1	/usr/lib/dracut/modules.d/70tpm2-totp/README
community/tpm2-totp 0.2.0-1	/usr/lib/dracut/modules.d/70tpm2-totp/cleanup-tpm2-totp.sh
community/tpm2-totp 0.2.0-1	/usr/lib/dracut/modules.d/70tpm2-totp/module-setup.sh
community/tpm2-totp 0.2.0-1	/usr/lib/dracut/modules.d/70tpm2-totp/show-tpm2-totp
community/tpm2-totp 0.2.0-1	/usr/lib/dracut/modules.d/70tpm2-totp/show-tpm2-totp.sh
```

Note: the zfs-utils AUR package installs files to both /usr/lib/initcpio/ and /usr/lib/dracut/ at the same time.

It seems perfectly fair to do mkinitcpio presets too. If dracut has the ability to register preset configurations, then I absolutely encourage the kernels to package both mkinitcpio and dracut presets.